plugins/woocommerce-blocks/assets/js/base/components/cart-checkout/totals/coupon/test/index.tsx

```diff
@@ -31,6 +32,9 @@ describe( 'TotalsCoupon', () => {
 			} );
 		} );
 		rerender( <TotalsCoupon instanceId={ 'coupon' } /> );
+
+		// TODO: Fix a recent deprecation of showSpinner prop of Button called in this component.
```

Comment:
Could you provide some context for this deprecation?

Reply:
You mean like add a link to the code where it's deprecated?

Reply:
I wanted to learn more about the reasons behind the deprecation. Could you mention them in the description PR? It should be announced at that level as well.

Reply:
@ralucaStan absolutely. I've outlined my reasoning in this comment: https://github.com/woocommerce/woocommerce/pull/45974#discussion_r1556928627

Do you feel this covers my reasoning well enough? If so I'll add it to the description.

Reply:
That comment shows that you aren't going to deprecate, but the discussion evolved and the decision was made to deprecate.

I'd be explicit in the description of the PR about the 2 deprecations introduced by this PR and the workarounds needed to accomplish the same behavior.

Reply:
I also recommend creating a ticket for actually deprecating the `showSpinner` property. In this case I think it's ok to recommend composability.

For the rest of the components from the project, I recommend documenting the nice to have improvements in new tickets and not deprecating parts of the API, even if they might make sense.

 I think it's important we focus on replacing the components and gathering nice to haves and suggestions for improving the APIs. This will mean little friction for our consumers and for us.

Reply:
I've updated the description to try explain both removal of icon and deprecation of showSpinner

Reply:
> I also recommend creating a ticket for actually deprecating the showSpinner property. In this case I think it's ok to recommend composability.

> For the rest of the components from the project, I recommend documenting the nice to have improvements in new tickets and not deprecating parts of the API, even if they might make sense.

I can see your view point, but why are we uncomfortable to deprecate anything? I think we should continue to improve things in place as we work on them personally, and deprecation has only been done where it's very low effort to do so. I think there is a tendency to write tickets for everything and dump them into a cooldown queue and never get to them? I think we should always be leaving the code better than we found it.

Reply:
I might have needed to be more explicit. I am not against deprecations. I was only asking to create a ticket for the actual removal of that code in the future. If we don't have a removal ticket then we risk living with that deprecation code hiding in the code base and not being intentional about removing it at some point.

Reply:
@ralucaStan Thanks I understand you now! 🙏🏻 

Here's an issue to track the removal: https://github.com/woocommerce/woocommerce/issues/46812

> For the rest of the components from the project, I recommend documenting the nice to have improvements in new tickets and not deprecating parts of the API, even if they might make sense.

This still sounds to me like you don't want us to do any actual deprecation for any more parts of this project though? I still think it would be a mistake to approach our code that way, but I'll keep this in mind as we work on the remaining components.

Reply:
> I'd be explicit in the description of the PR about the 2 deprecations introduced by this PR and the workarounds needed to accomplish the same behavior.

@ralucaStan yes sorry, I did intend to expand on what I had said in the comment. I've now expanded and given detail of things like workarounds to achieve the same result.

Reply:
I think this is resolved now.